relayer/src/chain/cosmos/batch.rs

```diff
@@ -105,33 +148,88 @@ async fn send_messages_as_batches(
         let response =
             send_tx_with_account_sequence_retry(config, key_entry, account, tx_memo, batch).await?;
 
-        if response.code.is_err() {
-            let events_per_tx = vec![IbcEvent::ChainError(format!(
-                "check_tx (broadcast_tx_sync) on chain {} for Tx hash {} reports error: code={:?}, log={:?}",
-                config.chain_id, response.hash, response.code, response.log
-            )); message_count];
-
-            let tx_sync_result = TxSyncResult {
-                response,
-                events: events_per_tx,
-                status: TxStatus::ReceivedResponse,
-            };
-
-            tx_sync_results.push(tx_sync_result);
-        } else {
-            let tx_sync_result = TxSyncResult {
-                response,
-                events: Vec::new(),
-                status: TxStatus::Pending { message_count },
-            };
-
-            tx_sync_results.push(tx_sync_result);
-        }
+        let tx_sync_result = response_to_tx_sync_result(&config.chain_id, message_count, response);
+
+        tx_sync_results.push(tx_sync_result);
     }
 
     Ok(tx_sync_results)
 }
 
+async fn sequential_send_messages_as_batches(
```

Comment:
I'm concerned about more code duplication here.
When contemplating refactoring for batching fixes respecting actual tx size, already there are two functions to care about, this adds a third one.

Could this be incorporated in the regular `send_messages_as_batches` under a dynamic flag?

Reply:
The core logic has already been abstracted into helper functions like `batch_messages`, `send_tx_with_account_sequence_retry`, and `wait_for_block_commits`. So functions like `sequential_send_messages_as_batches` only act as higher level functions that wire up the lower level functions. Such low level functions are already there to serve the code deduplication purpose.

Using a dynamic flag can obscure the logic that differentiates `send_messages_as_batches` from `sequential_send_messages_as_batches`. With them being separate functions, it is much more clear of how the behavior are different from one another.